integration-tests/gauntlet/gauntlet_solana.go

```diff
@@ -0,0 +1,514 @@
+	},
+		*sg.options,
+	)
+
+	if err != nil {
+		return "", err
+	}
+	sg.gr, err = sg.FetchGauntletJsonOutput()
+	if err != nil {
+		return "", err
+	}
+
+	return sg.gr.Responses[0].Contract, nil
+}
+
+// FetchTransmissions returns the last 10 transmissions
+func (sg *SolanaGauntlet) FetchTransmissions(ocrState string) ([]Transmission, error) {
+	_, err := sg.G.ExecCommand([]string{
+		"ocr2:inspect:responses",
+		ocrState,
+	},
+		*sg.options,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+	sg.gr, err = sg.FetchGauntletJsonOutput()
+	if err != nil {
+		return nil, err
+	}
+
+	return *sg.gr.Data.LatestTransmissions, nil
+}
+
+func (sg *SolanaGauntlet) DeployOCR2() (string, error) {
+	var err error
+
+	sg.AccessControllerAddress, err = sg.InitializeAccessController()
+	if err != nil {
+		return "", err
+	}
+
+	sg.BillingControllerAddress, err = sg.InitializeAccessController()
+	if err != nil {
+		return "", err
+	}
+
+	sg.StoreAddress, err = sg.InitializeStore(sg.BillingControllerAddress)
+	if err != nil {
+		return "", err
+	}
+	storeConfig := &StoreFeedConfig{
+		Store:       sg.StoreAddress,
+		Granularity: 1,
+		LiveLength:  10,
+		Decimals:    8,
+		Description: "Test feed",
+	}
+
+	sg.FeedAddress, err = sg.StoreCreateFeed(10, storeConfig)
+	if err != nil {
+		return "", err
+	}
+
+	_, err = sg.StoreSetValidatorConfig(sg.FeedAddress, 8000)
+	if err != nil {
+		return "", err
+	}
+
+	ocr2Config := &OCR2Config{
+		MinAnswer:     "0",
+		MaxAnswer:     "10000000000",
+		Transmissions: sg.FeedAddress,
+	}
+
+	sg.OcrAddress, err = sg.InitializeOCR2(sg.AccessControllerAddress, sg.BillingControllerAddress, ocr2Config)
+	if err != nil {
+		return "", err
+	}
+
+	storeWriter := &StoreWriterConfig{Transmissions: sg.FeedAddress}
+
+	_, err = sg.StoreSetWriter(storeWriter, sg.OcrAddress)
+	if err != nil {
+		return "", err
+	}
+
+	ocr2BillingConfig := &OCR2BillingConfig{
+		ObservationPaymentGjuels:  1,
+		TransmissionPaymentGjuels: 1,
+	}
+
+	_, err = sg.OCR2SetBilling(ocr2BillingConfig, sg.OcrAddress)
+	if err != nil {
+		return "", err
+	}
+
+	sg.ProposalAddress, err = sg.OCR2CreateProposal(2)
+	if err != nil {
+		return "", err
+	}
+	return "", nil
+}
+func (sg *SolanaGauntlet) ConfigureOCR2(onChainConfig common.OCR2OnChainConfig, offChainConfig common.OCROffChainConfig, payees common.PayeeConfig, proposalAccept common.ProposalAcceptConfig) error {
+	_, err := sg.ProposeOnChainConfig(sg.ProposalAddress, onChainConfig, sg.OcrAddress)
+	if err != nil {
+		return err
+	}
+	_, err = sg.ProposeOffChainConfig(sg.ProposalAddress, offChainConfig, sg.OcrAddress)
+	if err != nil {
+		return err
+	}
+	_, err = sg.ProposePayees(sg.ProposalAddress, payees, sg.OcrAddress)
+	if err != nil {
+		return err
+	}
+	_, err = sg.FinalizeProposal(sg.ProposalAddress)
+	if err != nil {
+		return err
+	}
+	_, err = sg.AcceptProposal(sg.ProposalAddress, "this is an testing only secret", proposalAccept, sg.OcrAddress)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
this would be a stretch goal to include:
* testing the payout mechanism works (harder)
* closing the feed works (this would be easier if the billing was just set to 0)

Reply:
if you close out the feed, you would also recover some SOL back to the original deployer

Reply:
Created task to followup on this as well